### ISSUE TITLE
Remove objc_sync_enter/objc_sync_exit

### ISSUE THREAD

**Issue by hamin** [hamin/FayeSwift#22]
I think we should just be able to use dispatch lock queues here like so:

```
let lockQueue = dispatch_queue_create("com.test.LockQueue", nil)
dispatch_sync(lockQueue) {
    // code
}
```

/cc @shams-ahmed

**Comment by shams-ahmed** [comment]
@hamin I agree only used `objc_sync_enter` due to its easy of use. ones the projects moved to swift 3 dispatch queue should be the way forward

**Comment by fengshi92** [comment]
May I ask when will this issue be addressed? It sometimes causes my urlSession delegate queue deadlock. Looking forward to hear from you!

**Comment by hamin** [comment]
@fengshi92 i haven't had a chance to look at this yet. Do you wanna take a shot at it and submit a PR? Since you're experiencing a deadlock issue right now your input would be super valuable

**Comment by fengshi92** [comment]
Due to legal reason of my company, I can't submit a pull request right now. Maybe later I could submit a fix. However I just want to let you know that the proposed method above resolved my problem. Thanks!

**Comment by hamin** [comment]
fixed by #27
